app/services/api/teachers/action.rb

```diff
@@ -0,0 +1,63 @@
+module API::Teachers
+  class Action
```

Comment:
I don't think we should break this out into `Defer < Action`, `Defer` service class on its own would be better. It'll be simpler without the abstraction.

Reply:
I don't mind either way for now; as long as the logic in `Action` will apply to all participant actions I think it makes sense to extract it, but if we want to pull it out into the parent class until we have all services in place in order to align `Defer` and `Withdraw` more easily I'm fine with that